File: functions/router/handler.go
package function

import (
	"fmt"
	"github.com/eclipse/paho.mqtt.golang"
	"log"
	"net/http"
	"os"
)

//"strings"

var (
	mqttBroker = os.Getenv("MQTT_URL")
	clientID   = os.Getenv("MQTT_CLIENTID")
)

// Fonction pour envoyer un message MQTT
func SendMQTT(messageJSON []byte, topic string) {
	mqttOpts := mqtt.NewClientOptions()
	mqttOpts.AddBroker(mqttBroker)
	mqttOpts.SetClientID(clientID)

	client := mqtt.NewClient(mqttOpts)
	if token := client.Connect(); token.Wait() && token.Error() != nil {
		log.Fatal(token.Error())
	}

	token := client.Publish(topic, 0, false, string(messageJSON))
	token.Wait()

	fmt.Printf("Message publié sur le topic %s: %s\n", topic, messageJSON)

	client.Disconnect(250)
}

// Middleware pour intercepter la requête HTTP
func Handle(w http.ResponseWriter, r *http.Request) {
	// Vous pouvez récupérer le message et le topic depuis la requête HTTP
	messageJSON := []byte(r.URL.Query().Get("message"))
	topic := r.URL.Query().Get("topic")

	if messageJSON == nil || topic == "" {
		http.Error(w, "Message ou topic manquant", http.StatusBadRequest)
		return
	}

	// Appeler la fonction pour envoyer le message MQTT
	SendMQTT(messageJSON, topic)

	// Répondre à la requête HTTP
	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusOK)
	response := fmt.Sprintf(`{"status": "Message envoyé sur le topic %s"}`, topic)
	w.Write([]byte(response))
}

File: functions/Fonctions test Proxy/function-2/handler.py
import paho.mqtt.client as mqtt
import json
import random
import os

# Configuration MQTT
MQTT_BROKER = os.getenv("MQTT_BROKER", "tcp://10.0.2.15")
MQTT_PORT = int(os.getenv("MQTT_PORT", 1883))
MQTT_TOPIC = "analyzedData"

def handle(req):
    client = mqtt.Client()
    client.connect(MQTT_BROKER, MQTT_PORT)

    # Générer des valeurs aléatoires
    data = {
        "temperature": round(random.uniform(10, 35), 2),
        "pressure": round(random.uniform(900, 1100), 2)
    }

    payload = json.dumps(data)
    client.publish(MQTT_TOPIC, payload)
    client.disconnect()

    return {
        "status": "Message published",
        "topic": MQTT_TOPIC,
        "data": data
    }